realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -131,6 +139,9 @@ public DynamicRealmObject(RealmModel obj) {
             case LIST:
                 return (E) getList(fieldName);
             default:
+                if (type.isDictionary()) {
```

Comment:
Oooff... I see why you are doing this. Personally, I would have added all the missing dictionary cases to the switch and have them do the same thing, just for clarity - but again, I see that it can be overkill, especially if we add multiple key types in the future 🤯 . If you don't want to do it, then I think it would be fair to add an explicit comment here mentioning why you handle them under the default case, or else it can be hard to reason about why it's there at a later point.

Reply:
I see your point of adding the cases. 

The switch case is not exhaustive because we have a default case, I added a generic case to forget about new types added under the dictionary (and set) as we have on lists. But I see your point that if they sit in the default case is not intuitive.

I am going to make the switch exhaustive without a default, to force writing any new supported case.